includes/ocos.h

```diff
@@ -57,55 +59,42 @@ struct OrtTensorDimensions : std::vector<int64_t> {
     return s;
   }
 
-  bool IsScalar() const{
+  bool IsScalar() const {
     return empty();
   }
 
-  bool IsVector() const{
+  bool IsVector() const {
     return size() == 1;
   }
 };
 
-
 template <typename... Args>
 class CuopContainer {
  public:
-#if defined(_MSC_VER) && !defined(__clang__)
-#pragma warning(push)
-#pragma warning(disable : 26409)
-#endif
-  CuopContainer() : ocos_list_({[]() { return new Args; }()...}) {
-    ocos_list_.push_back(nullptr);
+  CuopContainer() : op_instances_({[]() { return std::make_shared<Args>(); }()...}) {
+    ocos_list_.reserve(op_instances_.size());
+    std::transform(op_instances_.begin(), op_instances_.end(), std::back_inserter(ocos_list_),
+                   [](const std::shared_ptr<OrtCustomOp>& custom_op) { return custom_op.get(); });
   }
 
-  ~CuopContainer() {
-    if (0 < ocos_list_.size()) {
-      for (size_t i = 0; i < ocos_list_.size() - 1; i++) {
-          delete ocos_list_[i];
-      }
-    }
-    ocos_list_.clear();
-  }
-#if defined(_MSC_VER) && !defined(__clang__)
-#pragma warning(pop)
-#endif
-  const OrtCustomOp** GetList() {
-    return &const_cast<const OrtCustomOp*&>(ocos_list_.front());
+  const std::vector<const OrtCustomOp*>& GetCustomOps() const {
+    return ocos_list_;
   }
 
  private:
-  std::vector<OrtCustomOp*> ocos_list_;
+  std::vector<const OrtCustomOp*> ocos_list_;
+  std::vector<std::shared_ptr<OrtCustomOp>> op_instances_;  // use shared_ptr to capture type specific deleter
```

Comment:
This can hold any class derived from OrtCustomOp which is a C struct with no virtual destructor. I believe using shared_ptr will capture the deleter for the 'real' class and might be the only option in std for ensuring correct cleanup whilst storing the pointer in a generic vector containing OrtCustomOp instances.